api/src/main/java/org/pmiops/workbench/auth/DelegatedUserCredentials.java

```diff
@@ -0,0 +1,164 @@
+ * Researcher Workbench use case.
+ *
+ * <p>Example usage, for authorizing user requests to the Google Directory API:<br>
+ *
+ * <pre>
+ *   DelegatedUserCredentials delegatedCredentials = new DelegatedUserCredentials(
+ *     "service-account-with-dwd-enabled@project-name.iam.gserviceaccount.com",
+ *     "admin-gsuite-user@my-gsuite-domain.com",
+ *     DirectoryScopes.ADMIN_DIRECTORY_USERS);
+ *   Directory service = new Directory.Builder(new NetHttpTransport(), new JacksonFactory(), null)
+ *     .setHttpRequestInitializer(new HttpCredentialsAdapter(delegatedCredentials))
+ *     .build();
+ * </pre>
+ */
+public class DelegatedUserCredentials extends OAuth2Credentials {
+
+  static final String JWT_BEARER_GRANT_TYPE = "urn:ietf:params:oauth:grant-type:jwt-bearer";
+  static final String SERVICE_ACCOUNT_NAME_FORMAT = "projects/-/serviceAccounts/%s";
+  static final JsonFactory JSON_FACTORY = JacksonFactory.getDefaultInstance();
+  static final Duration ACCESS_TOKEN_DURATION = Duration.ofMinutes(60);
+
+  // The email of the service account whose system-managed key should be used to sign the JWT
+  // assertion which is exchanged for an access token. This service account:
+  // - Must have domain-wide delegation enabled for the target user's G Suite domain and scopes.
+  // - Does not need to be the same service account (SA) as the application default credentials
+  // (ADC)
+  //     service account. If they are different, the ADC account must have the Service Account Token
+  //     Creator role granted on this service account. See
+  //     https://cloud.google.com/iam/docs/creating-short-lived-service-account-credentials for more
+  //     details.
+  private String serviceAccountEmail;
+  // The full G Suite email address of the user for whom an access token will be generated.
+  private String userEmail;
+  // The set of Google OAuth scopes to be requested.
+  private List<String> scopes;
+  // The HttpTransport to be used for making requests to Google's OAuth2 token server. If null,
+  // a default NetHttpTransport instance is used.
+  private HttpTransport httpTransport;
+  // The IAM Credentials API client to be used for fetching a signed JWT from Google. If null,
+  // a default API client will be used.
+  private IamCredentialsClient client;
+
+  public DelegatedUserCredentials(
+      String serviceAccountEmail, String userEmail, List<String> scopes) {
+    super();
+    this.serviceAccountEmail = serviceAccountEmail;
+    this.userEmail = userEmail;
+    this.scopes = scopes;
+
+    if (this.scopes == null) {
+      this.scopes = new ArrayList<>();
+    }
+  }
+
+  @VisibleForTesting
+  public void setIamCredentialsClient(IamCredentialsClient client) {
+    this.client = client;
+  }
+
+  @VisibleForTesting
+  public void setHttpTransport(HttpTransport httpTransport) {
+    this.httpTransport = httpTransport;
+  }
+
+  /**
+   * Creates the set of JWT claims representing a service account `serviceAccountEmail` using
```

Comment:
These comments are great. It might help to have a readme going over the different auth types we have and in what situations we should use each one.